basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
+
+            int reference_line_start_index = waypoint_generation::get_nearest_point_index(reference_centerline, state);
+            int ending_line_end_index = waypoint_generation::get_nearest_point_index(ending_lane_centerline, ending_state);
+
+            lanelet::BasicLineString2d starting_centerline(reference_centerline.begin() + reference_line_start_index, reference_centerline.end());
+            lanelet::BasicLineString2d ending_centerline(ending_lane_centerline.begin(), ending_lane_centerline.begin() + ending_line_end_index);
+
+            //Convert to vector of BasicPoint2d
+
+                //Downsample centerlines
+            lanelet::BasicLineString2d downsampled_start_centerline;
+            downsampled_start_centerline.reserve(400);
+            downsampled_start_centerline = carma_utils::containers::downsample_vector(starting_centerline, 5);
+            std::vector<lanelet::BasicPoint2d> start_centerline;
+            for(int i = 0;i<starting_centerline.size();++i){
+                start_centerline.push_back(starting_centerline[i]);
+            }
+
+            lanelet::BasicLineString2d downsampled_end_centerline;
+            downsampled_end_centerline.reserve(400);
+            downsampled_end_centerline = carma_utils::containers::downsample_vector(ending_centerline, 5);
+            std::vector<lanelet::BasicPoint2d> end_centerline;
+            for(int i = 0; i<ending_centerline.size();++i){
+                end_centerline.push_back(ending_centerline[i]);
+            }
+            //If points are not the same size - ensure same size along both centerlines
+            if(start_centerline.size() != end_centerline.size())
+            {
+                //Resample centerlines to same length
+                auto centerlines = resample_linestrings_to_same_length(start_centerline, end_centerline);
+                start_centerline = centerlines.front();
+                end_centerline = centerlines.back();
+
+            }
+
+            //Create Trajectory geometry
+            double delta_step = 1.0 / start_centerline.size();
+
+            for (int i = 0; i < start_centerline.size(); i++)
+            {
+                lanelet::BasicPoint2d current_position;
+                lanelet::BasicPoint2d start_lane_pt = start_centerline[i];
+                lanelet::BasicPoint2d end_lane_pt = end_centerline[i];
+                double delta = delta_step * i;
+                current_position.x() = end_lane_pt.x() * delta + (1 - delta) * start_lane_pt.x();
+                current_position.y() = end_lane_pt.y() * delta + (1 - delta) * start_lane_pt.y();
+
+                centerline_points.push_back(current_position);
+            }
+
+            //Add points from following lanelet to provide sufficient distance for adding buffer
+            auto following_lanelets = wm->getMapRoutingGraph()->following(ending_lanelet, false);
+            if(!following_lanelets.empty()){
```

Comment:
Seems like this check is in the wrong place. The only purpose of the buffers is to support spline fitting, but the spline was already fit in the resample_linestrings_to_same_length method

Reply:
There are 3 instances of spline fitting here. The first 2 happen on the starting and target centerlines only to resample them. At this point we use a downsampled centerline which should always be able to fit through the spline.
If the centerlines are the same length, the spline fitting wouldn't happen.

At a later point, we resize the geometry points we are using for the trajectory, to be from the current state to the end.
Here we add a buffer, and then fit the spline through the actual lane change path/geometry points which the vehicle will use while lane changing.
I'm guessing a buffer would still be needed here?

Reply:
Can you point me to where this 3rd fit occurs?

Reply:
Just realized the third fit got overwritten, i'm adding it in now. It will happen in compose_lanechange_trajectory_from_path